apps/lectures/services/recommendation/content_based.py

```diff
@@ -0,0 +1,202 @@
+import logging
+import os
+import re
+from typing import List, Optional, Tuple
+
+import numpy as np
+import redis
+from dotenv import load_dotenv
+from numpy._typing import NDArray
+from sentence_transformers import SentenceTransformer
+
+from apps.lectures.models import CrawledLecture
+from apps.study_groups.models import GroupMember, StudyLecture
+from apps.users.models import User
+
+load_dotenv()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+ch.setFormatter(formatter)
+
+logger.addHandler(ch)
+
+redis_client = redis.Redis(
+    host=os.getenv("REDIS_HOST", "localhost"),
+    port=int(os.getenv("REDIS_PORT", 6379)),
+    db=int(os.getenv("REDIS_DB", 0)),
+)
+
+embedding_model = SentenceTransformer("all-MiniLM-L6-v2")
+
+STOPWORDS = {"강의", "수업", "소개", "배우기", "공부", "사용법"}
+
+EMBED_TTL = int(os.getenv("EMBED_TTL", 86400))
+EMBED_DIM = int(os.getenv("EMBED_DIM", 384))
+
+
+def lecture_embed_key(lecture_id: int) -> str:
```

Comment:
이것도 아마 캐시키일것같은데 명확하게 해주세요
```suggestion
def build_lecture_embed_cache_key(lecture_id: int) -> str:
```